generate_training_data_scripts/combine_core_noncore_rejected_data.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+import json
+import numpy as np
+import pandas as pd
+
+inspire_core_list_path = 'inspire_core_list.txt'
+inspire_noncore_list_path = 'inspire_noncore_list.txt'
+inspire_harvested_data_path = 'inspire_harvested_data.jsonl'
+inspire_core_data_path = 'inspire_core_records.jsonl'
+inspire_noncore_data_path = 'inspire_noncore_records.jsonl'
+save_path = 'inspire_data.df'
+
+# Get the rejected records from the harvested data
+inspire_core_arxiv_ids = np.loadtxt(inspire_core_list_path, delimiter='\n', dtype='<U14')
+inspire_noncore_arxiv_ids = np.loadtxt(inspire_noncore_list_path, delimiter='\n', dtype='<U14')
+
+def generate_rejected_data(harvested_data_path):
+    with open(harvested_data_path, 'r') as fd:
+        for line in fd:
+            try:
+                record = json.loads(line)
+                if not (record['arxiv_identifier'] in inspire_core_arxiv_ids) and \
+                        not (record['arxiv_identifier'] in inspire_noncore_arxiv_ids):
+                    del record['arxiv_identifier']
+                    yield record
+            except:
+                continue
+
+inspire_rejected_data = list(generate_rejected_data(inspire_harvested_data_path))
```

Comment:
I think you're missing the point here about using a generator. Let's discuss this.